app/models/host/managed.rb

```diff
@@ -916,10 +933,12 @@ def short_name_periods
   # but we should trigger it only for existing records and unless interfaces also changed (then validation is run
   # on them automatically)
   def trigger_nic_orchestration
-    self.primary_interface.valid? unless self.primary_interface.changed?
+    unless self.provision_interface.nil?
+      self.primary_interface.valid? unless self.primary_interface.changed?
 
-    if self.primary_interface != self.provision_interface && !self.provision_interface.changed?
-      self.provision_interface.valid?
+      if self.primary_interface != self.provision_interface && !self.provision_interface.changed?
+        self.provision_interface.valid?
+      end
     end
```

Comment:
Just use `self.primary_interface&.changed?` here, there is no need of introducing new block. The same for provision interface.